app/src/main/java/ch/epfl/qedit/backend/location/AndroidLocService.java

```diff
@@ -24,19 +24,19 @@ public boolean subscribe(LocationListener listener, int interval) {
         criteria.setAccuracy(Criteria.ACCURACY_FINE);
         criteria.setPowerRequirement(Criteria.POWER_MEDIUM);
         criteria.setAltitudeRequired(false);
-        criteria.setBearingRequired(false);
+        criteria.setBearingRequired(true);
         criteria.setCostAllowed(false);
         criteria.setSpeedRequired(false);
 
-        // If the application doesn't have the permission, we cannot subscribe
-        if (ActivityCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION)
-                        != PackageManager.PERMISSION_GRANTED
-                && ActivityCompat.checkSelfPermission(
-                                context, Manifest.permission.ACCESS_COARSE_LOCATION)
-                        != PackageManager.PERMISSION_GRANTED) return false;
+        // We need the permission manager to check the location permissions
+        PermissionManager permManager = PermManagerFactory.getInstance();
+
+        if (!permManager.checkPermission(context, Manifest.permission.ACCESS_FINE_LOCATION)
+                || !permManager.checkPermission(
+                        context, Manifest.permission.ACCESS_COARSE_LOCATION)) return false;
 
         // Otherwise, we subscribe to the location service
-        manager.requestLocationUpdates(interval, 10, criteria, listener, null);
+        manager.requestLocationUpdates(interval, 0, criteria, listener, null);
```

Comment:
This is the distance between the updates (the time is the first argument). But you are right, I need to change that, because 0 meters doesn't make sense. I don't really know what to put there to be honest, but it should be small enough so that the location is more precise.

Reply:
Oh yes I see. Of course the first one is the time interval.